### DISCUSSION TITLE
I want to use Standalone radio buttons material ui

### DISCUSSION THREAD

**Discussion by RajuTicvic** [react-hook-form/react-hook-form#2007] [Q&A] [answered]
I using material UI,
And I tried many ways to use the Standalone radio button but, all of my tries fails at the end.
Please help me with the working demo.
I don't want radio group option
What I want => like below
<Controller as={<Radio/>} name="test" value="1" control={control}/>
<Controller as={<Radio/>} name="test" value="2" control={control}/>
<Controller as={<Radio/>} name="test" value="3" control={control}/>

Thanks

**Comment by keiya01** [comment, marked as answer]
Please send code example with codesandbox.

**Reply by bluebill1049** [reply]
`Controller` need a unique name.

https://react-hook-form.com/api#Controller Take a look at example for MUI radio group: https://codesandbox.io/s/react-hook-form-v6-controller-qsd8r

**Reply by RajuTicvic** [reply]
How to use radio button without radio group

**Reply by bluebill1049** [reply]
just use native inputs then, Controller need unique name as simple as that.

**Reply by RajuTicvic** [reply]
No,I want to use material UI radio component.
here is my example, what I want like
https://codesandbox.io/s/react-hook-form-controller-5ct2o?file=/src/index.js

**Reply by bluebill1049** [reply]
<img width="1270" alt="Screen Shot 2020-07-06 at 3 29 09 pm" src="https://user-images.githubusercontent.com/10513364/86558691-70b67d80-bf9d-11ea-9444-c55535e74453.png">
your CSB have error and you can't Controller with the same `name`, that's why i was suggest to use a group component.

**Reply by RajuTicvic** [reply]
For design purpose, I want to use the radio in different places in the form, so do u have any other way for it. same name radio button

**Reply by bluebill1049** [reply]
you can have as many radio buttons as you want, but they will need to have unique name because it will be your submitted form data.

**Reply by RajuTicvic** [reply]
hmm, is there no way to use as I want? using react-hook-form

**Reply by bluebill1049** [reply]
if you having difficulty with the usage, you can always fall back to custom register, which is register input at useEffect and setValue.

```
useEffect(() => { register('test') }, [register])

setValue('test', 'xxx')
```

**Reply by RajuTicvic** [reply]
You mean we need to use this as custom through onchange radio option. Right?

**Reply by bluebill1049** [reply]
yes register the input and update via `setValue`

**Reply by RajuTicvic** [reply]
thanks bro

**Reply by zhushenhang** [reply]
I saw the MUI example you posted and used it.
I received a warning of missing defaultValue.
likes `missing in the 'defaultValue' prop of either its Controller`
when using radio, can i omit the defaultValue and ignore the warning?

**Reply by zhushenhang** [reply]
@bluebill1049 or i can set `defaultValue='' `  but I don't know if it will affect the code.
